File: 2020/F#/AdventOfCode/AdventOfCode/DayOne.fs
namespace AdventOfCode.Solutions

module DayOne = 
    let One (x: int list) : int =
        let res = [ for i in x do
                    for j in x do
                    if i + j = 2020 then i * j]
        res.Head

    let Two (x: int list) : int =
        let res = [ for i in x do
                    for j in x do
                    for z in x do
                    if i + j + z = 2020 then i * j * z ]
        res.Head

File: 2022/C#/AdventOfCode/DayOne.cs
namespace AdventOfCode
{
    internal static class DayOne
    {
        public static int One(List<int> calories)
        {
            var max = 0;
            var cur = 0;
            foreach (var val in calories)
            {
                if (val == int.MinValue)
                {
                    if (cur > max) { max = cur; }
                    cur = 0;
                }
                else
                {
                    cur += val;
                }
            }
            return max;
        }

        public static int Two(List<int> calories)
        {
            static void KeepLargest(int[] items, int item)
            {
                int curr = item;
                for (var i = 0; i < items.Length; i++)
                {
                    if (items[i] < curr)
                    {
                        (curr, items[i]) = (items[i], curr);
                    }
                }
            }

            var items = new int[3] { 0, 0, 0 };
            var cur = 0;

            foreach (var val in calories)
            {
                if (val == int.MinValue)
                {
                    KeepLargest(items, cur);
                    cur = 0;
                }
                else
                {
                    cur += val;
                }
            }

            return items.Sum();
        }
    }
}
